### DISCUSSION TITLE
Differences to nix-portable

### DISCUSSION THREAD

**Discussion by tennox** [nixie-dev/nixie#8] [Q&A] [answered]
Looks great! :heart_eyes: 
And a bit similar to [nix-portable](https://github.com/DavHau/nix-portable), I will check this out in more detail what the differences are, but if you already know it maybe you have an easy answer :thinking:

**Comment by Thesola10** [comment, marked as answer]
I won't lie, nix-portable was a big inspiration for Nixie.

I believe the primary difference between nix-portable and Nixie isn't so much technical, as it is one of intent.

Nix-portable aims to be as self-contained as possible. If you look at the size of the nix-portable file, for a single CPU architecture and system, it's really big. But the benefit is, more than Nix, you're getting a ready-to-go Nix store with a minimal work environment, all in a single file. This makes it a solid start for something like a live USB.

Nixie is different in that it relies a lot more on what the Nix package manager can do alone "in the wild". Since version 2.10, Nix can use a Nix store that isn't at the system root, much like what bwrap does in nix-portable. Nixie aims to be a developer tool for people who already have Nix installed, to provide a repository with a lightweight Nix wrapper script for contributors without Nix installed. This allows the resulting script to be much, much smaller as it expects to download its resources at runtime (though you can generate a script with the resources baked in).

As a developer tool, I thought Nixie should be targeting the most popular developer OSes, meaning it can also run on macOS (using an [ugly hack](https://github.com/nixie-dev/fakedir)), something that Nix-portable simply cannot do.

Also, Nixie's simpler design means it can be made to use the latest version of Nix and Nixpkgs. Nix-portable's build process really gets in the way of both Nix and Nixpkgs, which makes it way harder to update to the latest version.

In the end, DavHau himself hinted that Nixie is welcome as a "successor" to nix-portable in [this comment](https://github.com/DavHau/nix-portable/issues/42#issuecomment-1461331246), which also does an excellent job at explaining the more technical differences between the two.

Hope this helps!

**Reply by tennox** [reply]
That's a great explanation - thank you very much 🥰

Sounds like I will include Nixie in some of my projects 🤘
